### DISCUSSION TITLE
does duck db follow lazy evaluation stratergy?

### DISCUSSION THREAD

**Discussion by harjeet-glean** [duckdb/duckdb#10686] [Q&A]
what i mean is if i run duckdb.sql ("select * from read_parquet("s3:// file-with-partitions", hive_partitioning=1) where date='some-date' and some_column='value' limit 40").show()
Will duck db only fetch the data where data is some-date and some_column = value and steram the first 40 records,

Or it will bring the entre data of some date and then will do processing in memory and show me the output.
Will the same logic happens if I run a complex query involves joining of 10-11 tables doing coalesce, aggragtios etc.
//doing a poc

**Comment by soerenwolfers** [comment]
Yes, duckdb does its best to only fetch the data that's needed.
